src/classes/RD2_ScheduleService.cls

```diff
@@ -531,7 +531,7 @@ public without sharing class RD2_ScheduleService {
     public Date getNextDonationDateGreaterThan(RecurringDonationSchedule__c schedule, Date referenceDate) {
         Date firstValid = getFirstPossibleDonationDateFromStart(schedule);
 
-        if (firstValid >= referenceDate) {
+        if (firstValid > referenceDate) {
```

Comment:
I have serious concerns about this.  At the very least it requires rigorous testing.

Reply:
I just tested this and it seems to work.  The only concern I had was around the first payment date, and that works as expected, so I think this is okay.

Reply:
I had the same concern, it's honestly a little weird that it works, maybe we need a better understanding of `firstOccurrenceOnTimestamp` (that's where this Date ends up)